### PR TITLE
CXX-1973 Bump minimum required C driver

### PR DESCRIPTION
**Pull request by kevinAlbs**
1.17.0 is required for 4.4 support and Client-Side Field Level
Encryption.

Add a test to verify we can compile and test against the oldest
version of the C driver supported.

As a note: once this is merged, if the version of libmongoc you have installed locally is older than 1.17.0 (or reports as older) you may need to reinstall the C driver from the 1.17.0-beta2 release (or from master).

### CODE REVIEW

File: .mci.yml [hunk 1]
```diff
@@ -168,7 +168,7 @@ functions:
                         PREFIX=$(cygpath -m "$PREFIX")
                     fi
                     export GENERATOR="${cdriver_cmake_generator}"
-                    ./.evergreen/install_c_driver.sh ${MONGOC_VERSION|master}
+                    ./.evergreen/install_c_driver.sh ${mongoc_version|master}
```
Comment (kevinAlbs) [inline .mci.yml:171]: I changed the case for consistency with other expansions. I've been bitten before when treating an expansion as a bash variable and vice-versa.

File: .evergreen/build_example_projects.sh [hunk 2]
```diff
@@ -9,6 +9,9 @@ else
     DIR=shared
 fi
 
+. .evergreen/find_cmake.sh
+export CMAKE="$CMAKE"
```
Comment (kevinAlbs) [inline .evergreen/build_example_projects.sh:13]: The Ubuntu 18.04 hosts do not appear to have `cmake` in the path. `compile.sh` and `install_c_driver.sh` both use this script to discover the path to cmake, so this follows suit.

File: .mci.yml [hunk 3]
```diff
@@ -744,6 +734,22 @@ buildvariants:
...
+    # Test that we actually support building against the minium required C version claimed.
+    - name: ubuntu1804-min-libmongoc
+      display_name: "Ubuntu 18.04 with minimum libmongoc (MongoDB 4.4)"
+      batchtime: 1440 # 1 day
+      expansions:
+          build_type: "Debug"
+          tar_options: *linux_tar_options
+          cmake_flags: *linux_cmake_flags
+          mongodb_version: *version_latest
+          mongoc_version: "1.17.0-beta2"
```
Comment (kevinAlbs) [inline .mci.yml:746]: With this, the intent is if we introduce API into libmongoc, and start relying on that API in C++, we'll be alerted with a test failure if we do not bump the minimum C driver requirement.

File: src/mongocxx/CMakeLists.txt [hunk 4]
```diff
@@ -29,7 +29,7 @@ message ("mongocxx version: ${MONGOCXX_VERSION}")
 set(MONGOCXX_INLINE_NAMESPACE "v${MONGOCXX_ABI_VERSION}")
 set(MONGOCXX_HEADER_INSTALL_DIR "${CMAKE_INSTALL_INCLUDEDIR}/mongocxx/${MONGOCXX_INLINE_NAMESPACE}" CACHE INTERNAL "")
 
-set(LIBMONGOC_REQUIRED_VERSION 1.15.0)
+set(LIBMONGOC_REQUIRED_VERSION 1.17.0)
```
Comment (kevinAlbs) [inline src/mongocxx/CMakeLists.txt:32]: This is the only non-test change.

File: .mci.yml (hunk 1 above)
Comment (samantharitter) [reply .mci.yml:171]: Acknowledged

File: .evergreen/build_example_projects.sh (hunk 2 above)
Comment (samantharitter) [reply .evergreen/build_example_projects.sh:13]: Acknowledged

File: src/mongocxx/CMakeLists.txt (hunk 4 above)
Comment (samantharitter) [reply src/mongocxx/CMakeLists.txt:32]: I actually already needed to update my C driver installation, so seems good to make it "official" through the cmake

File: examples/projects/bsoncxx/cmake/static/build.sh [hunk 5]
```diff
@@ -3,20 +3,16 @@ set -o errexit
...
 set -o xtrace
 
-if [ -z "$BUILD_TYPE" ]; then
-    BUILD_TYPE=Release
-fi
-
-if [ -z "$CXX_STANDARD" ]; then
-    CXX_STANDARD=11
-fi
+BUILD_TYPE=${BUILD_TYPE:-Release}
```
Comment (samantharitter) [inline examples/projects/bsoncxx/cmake/static/build.sh:6]: TIL what the :- operator is!

**Pull request merged**